sdk/cosmos/azure-spring-data-cosmos-test/src/test/java/com/azure/spring/data/cosmos/core/CosmosTemplatePartitionIT.java

```diff
@@ -336,6 +325,36 @@ public void testPartitionedPaginationQuery() {
         PageTestUtils.validateLastPage(page, page.getContent().size());
     }
 
+    @Test
+    public void testPartitionedPaginationQueryWithOneOffset() {
+        cosmosTemplate.insert(TEST_PERSON_3, new PartitionKey(TEST_PERSON_3.getZipCode()));
+
+        final Criteria criteria = Criteria.getInstance(CriteriaType.IS_EQUAL, "firstName",
+            Collections.singletonList(FIRST_NAME), Part.IgnoreCaseType.NEVER);
+        final PageRequest pageRequest = CosmosPageRequest.of(1, 0, PAGE_SIZE_2,
+            null, Sort.by(ASC, "firstName"));
+        final CosmosQuery query = new CosmosQuery(criteria).with(pageRequest);
+
+        final Page<PartitionPerson> page = cosmosTemplate.paginationQuery(query, PartitionPerson.class, containerName);
+        assertThat(page.getContent().size()).isEqualTo(1);
+        PageTestUtils.validateLastPage(page, page.getContent().size());
```

Comment:
should we also validate which item is returned back?

Reply:
Done!